eth/catalyst/simulated_beacon.go

```diff
@@ -301,12 +319,14 @@ func (c *SimulatedBeacon) AdjustTime(adjustment time.Duration) error {
 	if parent == nil {
 		return errors.New("parent not found")
 	}
-	withdrawals := c.withdrawals.gatherPending(10)
+	withdrawals := c.withdrawals.Pop(10)
 	return c.sealBlock(withdrawals, parent.Time+uint64(adjustment/time.Second))
 }
 
+// RegisterSimulatedBeaconAPIs registers the simulated beacon's API with the
+// stack.
 func RegisterSimulatedBeaconAPIs(stack *node.Node, sim *SimulatedBeacon) {
-	api := &api{sim}
+	api := &simulatedBeaconAPI{sim: sim, doCommit: make(chan struct{}, 1)}
```

Comment:
Please relocate the simulated beacon api constructor to its own file

e.g.

```diff
func RegisterSimulatedBeaconAPIs(stack *node.Node, sim *SimulatedBeacon) {
-       api := &simulatedBeaconAPI{sim: sim, doCommit: make(chan struct{}, 1)}
-       if sim.period == 0 {
-               // mine on demand if period is set to 0
-               go api.loop()
-       }
        stack.RegisterAPIs([]rpc.API{
                {
                        Namespace: "dev",
-                       Service:   api,
+                       Service:   newSimulatedBeaconAPI(sim),
                        Version:   "1.0",
                },
        })
diff --git a/eth/catalyst/simulated_beacon_api.go b/eth/catalyst/simulated_beacon_api.go
index b26316f858..e98c14afca 100644
--- a/eth/catalyst/simulated_beacon_api.go
+++ b/eth/catalyst/simulated_beacon_api.go
@@ -30,6 +30,19 @@ type simulatedBeaconAPI struct {
        doCommit chan struct{}
 }
 
+// newSimulatedBeaconAPI initializes the simulated beacon api.
+func newSimulatedBeaconAPI(sim *SimulatedBeacon) *simulatedBeaconAPI {
+       api := &simulatedBeaconAPI{
+               sim:      sim,
+               doCommit: make(chan struct{}, 1),
+       }
+       if sim.period == 0 {
+               // mine on demand if period is set to 0
+               go api.loop()
+       }
+       return api

```

Reply:
done